src/PlusCommon/vtkPlusConfig.h

```diff
@@ -162,6 +162,9 @@ class vtkPlusCommonExport vtkPlusConfig : public vtkObject
   /*! Set device set configuration file name (if relative then the base is ProgramDirectory) */
   void SetDeviceSetConfigurationFileName(const std::string& aFilePath);
 
+  /*! Convenience function to create a device set configuration from a file */

```

Comment:
The return type of smart pointer is not clear enough that ownership need not be worried about? To me that's one of the fundamental benefits of using smart pointers.

Reply:
Is it just a comment that you would want added? Easily added. It is my assumption that if a method return smart pointer (vtk, igtl, std::shared_ptr, etc...) that memory ownership is not a concern.

Reply:
> The return type of smart pointer is not clear enough that ownership need not be worried about

Unfortunately, automatic casting from smart pointer to simple pointer has the disadvantage that you can assign the vtkSmartPointer to a simple pointer. However, a simple pointer does not keep the reference count at >0, so as soon as the statement is completed, the object is deleted. You don't get any compile-time error but it causes random application crashes depending on timing (if you use the object soon enough then the parts of the object you need may be still valid).

It could be solved by adding a "vtkSafePointer" to VTK, which would be a variant of vtkSmartPointer which could only be assigned to a vtkSmartPointer. However, it would also prevent simple one-line usage, where you don't need the memory after the statement is completed. So, the ideal solution would be to replace all pointers in VTK by smart pointers.

Anyway, for now a comment is enough, describe what exactly the caller has to do and why. If we can use the usual method of creating the VTK object before the method call then it is even better, because that it is really hard to use it incorrectly. It is more code (one additional line), but if it's only called from inside convenience functions then it should not be a problem.

Reply:
Oooh, because if people do `vtkSomeType* var = vtkSmartPointer<vtkSomeType>::New()` there is no compiler error, but you'll get a run-time error... right.

Hmmm, that sucks.